saga/sagaState.go

```diff
@@ -0,0 +1,292 @@
+package saga
+
+import "errors"
+import "fmt"
+
+/*
+ * Data Structure representation of the current state of the Saga.
+ */
+type SagaState struct {
+	sagaId string
+	job    []byte
+
+	taskStarted   map[string]bool
```

Comment:
added comments to each map.  The key is always the taskId.
  
TaskId has to be unique for Sagas to work just like SagaId must be unique for each saga.  In scoot, we will have unique TaskIds as part of the API to schedule work.